### ISSUE TITLE
symlinks are resolved when added

### ISSUE THREAD

**Issue by vm-wylbur** [iterative/dvc#10237]
## Description
Our projects use symlinks to track files output by one task that are inputs to other tasks. The symlinks are part of the project's documentation because they show the dependency structure among tasks. 

`dvc` seems to resolve symlinks. Is that always true or am I missing an option somewhere? 

### Reproduce

```
#!/bin/bash

# assume the existence of a remote at /var/repos/dvc/test w appropriate perms
DVCREPO=/var/repos/dvc/test
ls -la $DVCREPO
echo -e "\nempty remote, ready for data"

mkdir dir1
echo "1234567890" >dir1/data1.csv
echo "abcdefghijk" >dir1/data2.csv
ls -l dir1
echo "dir1 created with data files\n"

mkdir dir2
cd dir2
ln -sf ../dir1/data1.csv .
ln -sf ../dir1/data2.csv .
cd ..
ls -l dir2
echo -e "dir2 created with symlinks to data files\n"

echo "difference between files"
diff -qr --no-dereference dir1 dir2

dvc init --no-scm -q

dvc remote add -f myremote $DVCREPO
dvc remote default myremote
cat .dvc/config

dvc add dir1 dir2
dvc commit
dvc push
echo "data pushed to remote repo"

ls -lR $DVCREPO
cat $DVCREPO/files/md5/7e/47cdf27c6023e4f6bfb6cb28b26de7.dir

rm -r dir1 dir2
echo "after removing dir1 and dir2"
ls -la .

echo -e "\npulling from repo"
dvc pull
ls -l dir1
ls -l dir2
diff -qr --no-dereference dir1 dir2

set +x
# done
```
**output**
```
../dvc-test.sh
total 8
drwxrws---+ 2 pball test 4096 Jan 13 01:55 .
drwxrws---+ 4 pball svn  4096 Jan 13 01:55 ..

empty remote, ready for data
total 8
-rw-r--r-- 1 pball svn 11 Jan 13 01:55 data1.csv
-rw-r--r-- 1 pball svn 12 Jan 13 01:55 data2.csv
dir1 created with data files\n
total 0
lrwxrwxrwx 1 pball svn 17 Jan 13 01:55 data1.csv -> ../dir1/data1.csv
lrwxrwxrwx 1 pball svn 17 Jan 13 01:55 data2.csv -> ../dir1/data2.csv
dir2 created with symlinks to data files

difference between files
File dir1/data1.csv is a regular file while file dir2/data1.csv is a symbolic link
File dir1/data2.csv is a regular file while file dir2/data2.csv is a symbolic link
[core]
    no_scm = True
    remote = myremote
['remote "myremote"']
    url = /var/repos/dvc/test
100% Adding...|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|2/2 [00:00, 38.17file/s]
3 files pushed
data pushed to remote repo
/var/repos/dvc/test:
total 4
drwxr-xr-x+ 3 pball test 4096 Jan 13 01:55 files

/var/repos/dvc/test/files:
total 4
drwxr-xr-x+ 5 pball svn 4096 Jan 13 01:55 md5

/var/repos/dvc/test/files/md5:
total 12
drwxr-xr-x+ 2 pball svn 4096 Jan 13 01:55 7c
drwxr-xr-x+ 2 pball svn 4096 Jan 13 01:55 7e
drwxr-xr-x+ 2 pball svn 4096 Jan 13 01:55 e2

/var/repos/dvc/test/files/md5/7c:
total 4
-r--r--r-- 1 pball svn 11 Jan 13 01:55 12772809c1c0c3deda6103b10fdfa0

/var/repos/dvc/test/files/md5/7e:
total 4
-r--r--r-- 1 pball svn 138 Jan 13 01:55 47cdf27c6023e4f6bfb6cb28b26de7.dir

/var/repos/dvc/test/files/md5/e2:
total 4
-r--r--r-- 1 pball svn 12 Jan 13 01:55 71dc47fa80ddc9e6590042ad9ed2b7
[{"md5": "7c12772809c1c0c3deda6103b10fdfa0", "relpath": "data1.csv"}, {"md5": "e271dc47fa80ddc9e6590042ad9ed2b7", "relpath": "data2.csv"}]after removing dir1 and dir2
total 24
drwxr-xr-x 3 pball svn 4096 Jan 13 01:55 .
drwxr-xr-x 3 pball svn 4096 Jan 13 01:49 ..
-rw-r--r-- 1 pball svn   98 Jan 13 01:55 dir1.dvc
-rw-r--r-- 1 pball svn   98 Jan 13 01:55 dir2.dvc
drwxr-xr-x 4 pball svn 4096 Jan 13 01:55 .dvc
-rw-r--r-- 1 pball svn  139 Jan 13 01:55 .dvcignore

pulling from repo
Collecting                                                                                                                                   |6.00 [00:00,  191entry/s]
Fetching
Building workspace index                                                                                                                     |0.00 [00:00,    ?entry/s]
Comparing indexes                                                                                                                           |7.00 [00:00, 1.93kentry/s]
Applying changes                                                                                                                             |4.00 [00:00,   691file/s]
A       dir2/
A       dir1/
2 files added
total 8
-rw-r--r-- 1 pball svn 11 Jan 13 01:55 data1.csv
-rw-r--r-- 1 pball svn 12 Jan 13 01:55 data2.csv
total 8
-rw-r--r-- 1 pball svn 11 Jan 13 01:55 data1.csv
-rw-r--r-- 1 pball svn 12 Jan 13 01:55 data2.csv
```

### Expected

I need to preserve the symlinks. 

### Environment information

```
 $ dvc doctor
DVC version: 3.33.4 (conda)
---------------------------
Platform: Python 3.11.5 on Linux-5.15.0-91-generic-x86_64-with-glibc2.35
Subprojects:
	dvc_data = 2.24.0
	dvc_objects = 2.0.1
	dvc_render = 1.0.0
	dvc_task = 0.3.0
	scmrepo = 1.6.0
Supports:
	http (aiohttp = 3.8.5, aiohttp-retry = 2.8.3),
	https (aiohttp = 3.8.5, aiohttp-retry = 2.8.3)
Config:
	Global: /home/pball/.config/dvc
	System: /etc/xdg/dvc
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/mapper/VGscott-home
Caches: local
Remotes: local
Workspace directory: ext4 on /dev/mapper/VGscott-home
Repo: dvc (no_scm)
Repo.site_cache_dir: /var/tmp/dvc/repo/625fc0929c52fe30feae4440628c3033
```

## note
I suspect that this is the intended behavior, but I have been confused by [this paragraph in the docs](https://dvc.org/doc/command-reference/add#add-symlink) which says "`dvc add` supports symlinked files as targets..." If `dvc` supports symlinks by resolving them, then I don't think `dvc` is _supporting_ symlinks, and perhaps the docs could be clarified. 

However, if I'm wrong (which I hope I am!!), maybe someone could point me to the options necessary to preserve the symlinks in the dvc repository? thank you.

**Comment by vm-wylbur** [comment]
I've found other versions of this concern, like #9971 and [this discussion](https://discuss.dvc.org/t/symlink-semantics-in-tracked-directories/1645/2). If I understand correctly, this means for our project, we will need to dvc-track **files** not **directories**. That way we can keep symlinks out of dvc and track them only with git. That's a little more work in our flow, but manageable. I think the paragraph I referenced in the docs should be amended.
